IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+
+
+## Peg-In using Destination-Mapped Federation Addresses
+
+It’s also possible to create a Bitcoin address that automatically assigns the ownership of the ordinal in Rootstock to a specific RSK address or contract. This is called destination-mapped federation addresses (DMFA). This can be used to automatically mint an NFT when the ordinal is transferred to Rootstock. 
+
+This requires creating a special unique single-use reception address for the ordinal. The address is created with the following method:
+
+
+`void registerBtcTransactionWithOrdinalsDMFA(
+   bytes rskTx,
+   bytes btcTxSerialized,
+   uint32 height,
+   bytes pmtSerialized,
+   address rskOwner);`
+
+The field rskOwner specifies the owner of the ordinal.
+
+## Changing the Ownership of an Ordinal
+
+To query who owns an ordinal, the following method is added:
+
+`address getOrdinalOwner(uint256 tId, int index)`
+
+The method will return 0x00... (the zero 20-byte address) if the ordinal does not exist.
+
+To change ownership of an ordinal, the following method must be called by the owner:
+
+`void transferOrdinal(uint256 tId, int index, address newOwner)`
+
+The method does nothing if the ordinal does not exist.
+
+## Ordinal Peg-outs
+
+To send an ordinal, the following method must be called:
+
+`void releaseOrdinal(uint256 txId, uint256 index, bytes btcAddress)`
+
+This method will queue the release of an ordinal, sending it to the btcAddress specified. The caller must be the owner of the ordinal. Every 1 hour the ordinals that are queued to be released will be processed. The bridge may attempt to create a single transaction containing all ordinal UTXOs, and all ordinal outputs, plus a single additional input used for fee payments, and a single additional output used for change.
+
+Ordinals can be sent to any Bitcoin address.
+
+## Paying Parking Fees
+Each ordinal parked stores the date when the last annual parking fee was paid. Initially this date corresponds to the date of parking. The parking expires one year later. If the parking expires and the fee has not been paid, anyone will be able to claim the ordinal by paying its parking fee. If a bridge migration event occurs, all parked ordinals that have their fee unpaid will not be migrated, and they will be lost forever.
+
+To know how much must be paid to park an ordinal, the Bridge provides the following method: 
+
+`uint getOrdinalsAnnualParkingFee()`
+
+This method returns the fee that ordinals must pay annually for being parked. The amount is given in rBTC weis.
+
+To pay the parking fee of an ordinal, the Bridge provides the following method:
+
+`void payParkingExtensionFee(uint256 tId, int index)`
+
+This method extends the parking time of the ordinal to a year starting from the current date.
+ 
+To know exactly how much you need to pay to extend the parking time, the Bridge provides the following method:
+ 
+`uint256 getOrdinalParkingExtensionFee(uint256 tId, int index)`
```

Comment:
Is this value different from `getOrdinalsAnnualParkingFee`? It's still the fee for parking the ordinal for a year, isn't it?

Reply:
This is the extension cost. If currently the expiration is in 6 months, then the value returned will be half of the annual cost, because 6 months have already been paid and only 6 months remain to be paid. Remember that the rent cannot be paid in advance for more than a year.

Reply:
clarified